### ISSUE TITLE
Extend free text search to find matches in resource texts

### ISSUE THREAD

**Issue by acn-sbuad** [Altinn/altinn-storage#8]
## Description
By extending the search functionality to also find matches in the resource text of an application we would have a more flexible solution.

## Considerations
- Do our users miss this functionality or is it simply nice to have and cool to implement?
- Performance of the search API cannot be significantly decreased due to this change.
- Do we require a minimum number of characters before doing a search?
- General search words might result in a large number of matches. 

## Acceptance criteria
> Describe criteria here (i.e. What is allowed/not allowed (negative tesing), validations, error messages and warnings etc.)

## Analysis

**[ Complexity: medium]**  - once a text match is found we need to send a new request to identify which instances are related to the application before we can return a result.

**[ Performance ]**  - A bit worried about the performance here. We would have to run tests to see how it can be optimized. 

- As all resource texts might not be cached in Storage, a query towards CosmosDB with each free text search is required. 

- For a given instance owner or set of instance owners we would only need to search the text resources related to 
applications that they actually have instances of. 

- It is possible to identify which application has a resource text that matches free text. 
If variables in text are used, it will not be possible to search on the variable values. 

- To limit the search are a filter on the users language in the messagebox could be set on the query as well

- It is suggested that we search the value of the text and not the text key. 

```sql
SELECT texts.id
    FROM texts
    JOIN r IN texts.resources 
    where CONTAINS(r["value"],'Search word', true)
    and texts.language="nb"
```

## Specification tasks

- [ ] Development tasks are defined
- [ ] Test design / decide test need

## Development tasks
> Add tasks here

## Definition of done
Verify that this issue meets [DoD](https://confluence.brreg.no/display/T3KP/Definition+of+Done#DefinitionofDone-DoD%E2%80%93utvikling) (Only for project members) before closing.

- [ ] Documentation is updated (if relevant)
  - [ ] Technical documentation (docs.altinn.studio)
  - [ ] User documentation (altinn.github.io/docs)
- [ ] QA
- [ ] Manual test is complete (if relevant)
- [ ] Automated test is implemented (if relevant)
- [ ] All tasks in this userstory are closed (i.e. remaining tasks are moved to other user stories or marked obsolete)

**Comment by SandGrainOne** [comment]
Texts are being moved to a separate micro service.
